pgmpy/models/DoubleMLRegressor.py

```diff
@@ -0,0 +1,188 @@
+                    ]
+                except Exception:
+                    adj = []
+        return treat, adj
+
+    def fit(self, X, y, sample_weight: Optional[np.ndarray] = None):
+
+        X_arr, y_arr = check_X_y(
+            X, y, accept_sparse=False, ensure_2d=True, force_all_finite=True
+        )
+
+        X = X if isinstance(X, pd.DataFrame) else self._as_dataframe(X_arr)
+        feature_names = list(X.columns)
+
+        self.n_features_in_ = X_arr.shape[1]
+        self.feature_names_in_ = np.array(feature_names, dtype=object)
+
+        df = X.copy()
+        df["outcome"] = np.asarray(y_arr).ravel()
+
+        treat_col, adj_cols = self._resolve_treatment_and_adjustments(df)
+        design_cols = [treat_col] + [c for c in adj_cols if c != treat_col]
+        y_vec = df["outcome"].to_numpy(dtype=float)
+
+        # clone nuisance learners and fit (full-sample, simple) ----
+        ml_g = clone(self.estimator_g)
+        ml_m = (
+            clone(self.estimator_m)
+            if self.estimator_m is not None
+            else clone(self.estimator_g)
+        )
+
+        adj_cols_for_nuisance = [c for c in design_cols if c != treat_col]
+        if len(adj_cols_for_nuisance) == 0:
+            X_for_g = np.empty((len(y_vec), 0))
+        else:
+            X_for_g = df[adj_cols_for_nuisance].to_numpy(dtype=float)
+
+        if X_for_g.shape[1] == 0:
+            g_hat = np.repeat(y_vec.mean(), len(y_vec))
+            self.estimator_g_ = None
+        else:
+            ml_g.fit(X_for_g, y_vec)
+            self.estimator_g_ = ml_g
+            g_hat = ml_g.predict(X_for_g)
+            g_hat = np.asarray(g_hat).ravel()
+
+        # fit m (treatment mechanism)
+        t_vec = df[treat_col].to_numpy(dtype=float)
+        if X_for_g.shape[1] == 0:
+            m_hat = np.repeat(t_vec.mean(), len(t_vec))
+            self.estimator_m_ = None
+        else:
+            ml_m.fit(X_for_g, t_vec)
+            self.estimator_m_ = ml_m
+            # if classifier, prefer predict_proba for propensity
+            if hasattr(ml_m, "predict_proba"):
+                try:
+                    probs = ml_m.predict_proba(X_for_g)
+                    # if binary, take second column
+                    if probs.ndim == 2 and probs.shape[1] >= 2:
+                        m_hat = probs[:, 1]
+                    else:
+                        m_hat = probs.ravel()
+                except Exception:
+                    m_hat = ml_m.predict(X_for_g)
+            else:
+                m_hat = ml_m.predict(X_for_g)
+            m_hat = np.asarray(m_hat).ravel()
+
+        self.g_hat_ = g_hat
+        self.m_hat_ = m_hat
+
+        # orthogonalization
+        y_res = y_vec - self.g_hat_
+        t_res = t_vec - self.m_hat_
+
+        # regress y_res on t_res (with intercept)
+        X_res = np.column_stack([np.ones(len(t_res)), t_res])
+        theta_coef, *_ = np.linalg.lstsq(X_res, y_res, rcond=None)
```

Comment:
Would suggest using the linear regression implementation from statsmodels, it will also give us summary stats for the fit.